BORGENING! Borg changes and QOLs

Aliceee2ch opened the pull request:
## About The Pull Request

Adds a few new items and upgrades for different cyborg modules or reworks some of them. 

<p align="center">
<img width="608" height="199" alt="image" src="https://github.com/user-attachments/assets/ba44742f-984b-4988-af4d-e06cf66d3296" />
</p>
pic unrelated :)

### Main changes:

### Medical model
**Medical cyborg** gets chembag by default and a new bluespace syringe upgrade!
<p align="center">
<img width="267" height="83" alt="image" src="https://github.com/user-attachments/assets/4636688f-a3bf-4db8-b0c7-75089aeb4f54" />
</p>

### Engineering model
**Engineering cyborg** recieved a few new features. Now you can put your regular RPED inside of a cyborg, but mind that it has way less storage, than RPED upgrade from robotics. Robotics RPED upgrade were renamed into expanded RPED and now cost the same amount of material to produce it as BS RPED.
<p align="center">
<img width="198" height="117" alt="image" src="https://github.com/user-attachments/assets/e614a9b0-41ad-4815-8eb6-f627c2bf77cc" />
</p>

Worth to mention adding decal painter with own sprite (yes it uses cell instead of regular toner)
<p align="center">
<img width="827" height="498" alt="image" src="https://github.com/user-attachments/assets/2eb36651-0ee2-4c9b-81ef-f8c3a96966a8" />
</p>

<p align="left">
Circuit manipulator was renamed to engineering apparatus, as its can no longer hold only circuits. Additionally, it can also hold tubes/bulbs now.
</p>

### Mining model
**Mining cyborg** now has mesons removed and remade into a new toggle button, you dont need to use one of your modules just to be able to see through darkness of lavaland caves.
<p align="center">
<img width="248" height="173" alt="image" src="https://github.com/user-attachments/assets/e85a00ec-f373-43d0-9ce9-7ec573029af3" />
</p>

**Mining cyborg** got a new shield module. It helps you fighting lavaland fauna (..or annoying humans, if youre evil) in low pressure by absorbing 50% of incoming damage when activated. Takes some amount of your charge per absorbed hit.
Example:
![2025-07-30 15-05-38](https://github.com/user-attachments/assets/0337a3b2-b9ef-4e27-a464-e3bc0a6a77d8)

## Why It's Good For The Game

I feel big lack of some cyborg modules or even features. This PR is supposed to fix missing gap, aswell as rebalance some of existing models.

## Changelog


:cl:
qol: Added new helpful modules, such as decal painter for engineering and chembag for medical borg.
add: Added miner cyborg shield module. In lavaland (low) pressure, it protects you from 50% of incoming damage in exchange of your cell charge.
add: Added BS syringe upgrade for medical cyborg.

/:cl:

Aliceee2ch committed:
yeah im a batman

Kurgis commented:
Just since it wasn't mentioned, installing an expanded RPED even after a regular RPED was installed won't run in to incompatibility issues will it? If I may offer one suggestion, janitor borgs constantly bring floor buffer upgrades from the service lathe to Robotics in order to ask for an upgrade installation, perhaps there's a possible addition there? It seems natural enough if many separate players already assume it's a feature.

I disagree with the research changes, this is just going to make it take longer for borgs to get their upgrades overall now due to how NT Frontier works. There should be some delay in them being able to select the research, of at least ~4 to ~5 minutes assuming Science is active also queuing up nodes. Otherwise why have those nodes in NT Frontier at all if they're impossible to complete in time? It takes ~6 minutes within roundstart usually to complete plasma, N2O, and BZ gas shells, but they can rush the console now and queue it for 3 minutes within roundstart while now also completely shafting the mining borg which provides actual material value.

Mining borg changes, thank, you, so much. Most annoying thing about the mining module, and this is a flat-out improvement for something in such a precarious position. They need all the help they can get, and being stuck with goggles in your module bar wasn't one of them.

moocowswag commented:
the mesons toggle is the only part that feels like power creep (but its mainly pve power creep) 

The rapid part exchange upgrade no longer being arbitrarily locked behind tens of thousands of points worth of research is ideal, building machines is a core part of construction and the fact that the borg model that is meant for construction doesn't get access to it most rounds is kind of insane and felt like an oversight.

Dawnseer commented:
> " Additionally, it can also hold tubes/bulbs now." this is the janitor borgs job

I'd argue that no, it's not the janiborgs job. The janiborgs job is to replace broken bulbs and tubes. The engineering borg can build lights, but is currently unable to actually put bulbs and tubes in them, which is stupid.

Tostah commented:
It's a bit pedantic, but this is a bit more than a QOL and essentially a pretty large balance change. The changes themselves are really good though, been waiting a while for some of these.

sippykot commented:
> ```css
> >QOL PR
> >Look inside
> >power creep
> ```

```css
>tg borg code
>look inside
>salt
```

Aliceee2ch committed:
a

Aliceee2ch committed:
revert tech nodes + plasma cutter upgrade

Aliceee2ch committed:
revert this too oops

Aliceee2ch committed:
hm

Aliceee2ch committed:
more

Aliceee2ch committed:
oops!

Aliceee2ch committed:
temporary

Aliceee2ch committed:
Merge branch 'master' into borg-qols

LemonInTheDark commented:
I don't want to go change by change but I think borg features work best when they are distinct from the job they're mirroring. like, I like the shield module a lot, cause it's unique and works as armor but cool. I don't really like the cyber cutter cause it's just the same tool but strapped to a borg (goes for a few of the things here tho I realize some of this is just a matter of reasonability.)

To be specific about a few other things, do not give borgs a universal translator that separation is important I think.
I also don't think engi needs all you're giving it, it's ok for the borg to not be able to do everything people can, that distinction is useful IMO.

Aliceee2ch committed:
hello beautiful

Aliceee2ch committed:
Merge branch 'master' into borg-qols

Aliceee2ch commented:
discussed some things with Lemon in discord and i removed janitor, service ideas and miner borg cutter. (i dont really have ideas for janiborg+service rn so i will make separate pr next time, i have service rework in mind already). updated body btw

Aliceee2ch committed:
123

Aliceee2ch committed:
Merge branch 'borg-qols' of https://github.com/Aliceee2ch/2chstation into borg-qols

LemonInTheDark commented on code/modules/mob/living/silicon/robot/robot.dm:
```diff
@@ -312,6 +312,12 @@
 		eye_lights.icon = icon
 		add_overlay(eye_lights)
 
+	var/obj/item/shield_module/shield_module = locate(/obj/item/shield_module) in src
+	if(shield_module && shield_module.active)
+		add_overlay(shield_module.shield_overlay)
+	else if (shield_module && !shield_module.active)
+		cut_overlay(shield_module.shield_overlay)
```
oh my god I hate  borg code

LemonInTheDark commented on code/modules/mob/living/silicon/robot/robot_defense.dm:
```diff
@@ -517,3 +535,25 @@ GLOBAL_LIST_INIT(blacklisted_borg_hats, typecacheof(list( //Hats that don't real
+		balloon_alert(borg, "absorbed!")
+		borg.cell.use(damage * (STANDARD_CELL_CHARGE / 15), force = TRUE)
+		damage *= 0.5
+	. = ..()
```
return here not . = (clearer)

LemonInTheDark commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -743,6 +769,20 @@
 		"Lavaland Miner" = list(SKIN_ICON_STATE = "miner"),
 	)
+	var/datum/weakref/night_vision_ref
+	var/obj/item/shield_module/shield
```
this var is unused? (old copypasta?)

LemonInTheDark commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -631,15 +631,36 @@
+/obj/item/borg/upgrade/rped/action(mob/living/silicon/robot/borg, mob/living/user = usr)
+	. = ..()
+	if(!.)
+		return .
```
if you're gonna do return . here you should be consistent in your pr

LemonInTheDark commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -937,6 +978,34 @@
+	name = "Toggle Thermal Night Vision"
+	button_icon = 'icons/mob/actions/actions_mecha.dmi'
+	button_icon_state = "meson"
+	var/list/lighting_color_cutoffs
```
similar copypasta?

LemonInTheDark commented on code/modules/mob/living/silicon/robot/robot_model.dm:
```diff
@@ -937,6 +978,34 @@
+/datum/action/cooldown/borg_thermal/Activate()
+	if(usr.sight & SEE_TURFS)
+		usr.clear_sight(SEE_TURFS|SEE_MOBS)
+		usr.lighting_cutoff = 0
+	else
+		usr.add_sight(SEE_TURFS|SEE_MOBS)
+		usr.lighting_cutoff = 30
+	usr.sync_lighting_plane_cutoff()
```
why use a static (used for rb and b) value and not one of the lists?

LemonInTheDark commented on code/modules/research/designs/mechfabricator_designs.dm:
```diff
@@ -1652,13 +1652,14 @@
 	build_path = /obj/item/borg/upgrade/rped
 	materials = list(
-		/datum/material/iron =SHEET_MATERIAL_AMOUNT*5,
+		/datum/material/iron = SHEET_MATERIAL_AMOUNT*7.5,
 		/datum/material/glass =SHEET_MATERIAL_AMOUNT * 2.5,
+		/datum/material/silver = HALF_SHEET_MATERIAL_AMOUNT*2.5
```
spacing around *s pls (or failing that do me a favor and make this list self consistent, that's most of my concern in this case)

LemonInTheDark commented on code/modules/research/designs/mechfabricator_designs.dm:
```diff
@@ -1718,6 +1719,22 @@
+	build_type = MECHFAB
+	build_path = /obj/item/borg/upgrade/bs_syringe
+	materials = list(
+		/datum/material/iron =SHEET_MATERIAL_AMOUNT,
+		/datum/material/plasma =HALF_SHEET_MATERIAL_AMOUNT,
+		/datum/material/diamond =HALF_SHEET_MATERIAL_AMOUNT,
+		/datum/material/bluespace =SMALL_MATERIAL_AMOUNT*5
```
oh my god this is the standard in this file isn't it

Aliceee2ch committed:
ze reviews
